internal/core/software_renderer.rs

```diff
@@ -1180,28 +1180,28 @@ impl<B: target_pixel_buffer::TargetPixelBuffer> RenderToBuffer<'_, B> {
     }
 
     fn process_texture_impl(&mut self, geometry: PhysicalRect, texture: SceneTexture<'_>) {
-        if !self.buffer.draw_texture(
-            geometry.origin.x,
-            geometry.origin.y,
-            geometry.size.width,
-            geometry.size.height,
-            target_pixel_buffer::Texture {
-                bytes: texture.data,
-                pixel_format: texture.format,
-                pixel_stride: texture.pixel_stride,
-                width: texture.source_size().width as u16,
-                height: texture.source_size().height as u16,
-                delta_x: texture.extra.dx.0,
-                delta_y: texture.extra.dy.0,
-                source_offset_x: texture.extra.off_x.0,
-                source_offset_y: texture.extra.off_y.0,
-            },
-            texture.extra.colorize.as_argb_encoded(),
-            texture.extra.alpha,
-            texture.extra.rotation,
-            CompositionMode::default(),
-        ) {
-            self.foreach_region(&geometry, |buffer, rect, extra_left_clip, extra_right_clip| {
+        self.foreach_region(&geometry, |buffer, rect, extra_left_clip, extra_right_clip| {
+            if !buffer.draw_texture(
+                geometry.origin.x,
+                geometry.origin.y,
+                geometry.size.width,
+                geometry.size.height,
```

Comment:
This needs to be the region, not the full geometry

Reply:
You also need to adjust the source_offset_x/y

Reply:
Argh, thanks. 

> You also need to adjust the source_offset_x/y

How does it need to be adjusted?

Reply:
Depends on how you want the `draw_texture` function to work. But the implementation needs to know what pixel of the image goes where.
So if you want the function to work the same as our current draw_function, you can increment the source_offset by the difference between the geometry origin and the region origin. I think that would work.  the extra clip thing may also help.